clients/src/test/java/org/apache/kafka/clients/consumer/internals/FetcherTest.java

```diff
@@ -1669,6 +1669,35 @@ protected boolean shouldRetain(RecordBatch recordBatch, Record record) {
     }
 
     @Test
+    public void testUpdatePositionOnEmptyBatch() {
```

Comment:
Hmm I think you meant to comment on `testEmptyHeader`. This test is used to validate that the fetcher change works correctly.